### PR TITLE
Simplify host triplet passing code

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Review by nrc — COMMENTED** [review]
Looks good other than the question about the `rerun` directive

File: src/rustup-dist/build.rs [hunk 1]
```diff
@@ -1,15 +1,6 @@
...
 
 fn main() {
-    let out_dir = PathBuf::from(env::var_os("OUT_DIR").unwrap());
     let target = env::var("TARGET").unwrap();
-
-    File::create(out_dir.join("target.txt"))
-        .unwrap()
-        .write_all(target.as_bytes())
-        .unwrap();
-    println!("cargo:rerun-if-changed=build.rs");
```
Comment (nrc) [inline src/rustup-dist/build.rs]: Looks like my comment was lost: I thought we would need to keep this directive in order to avoid unnecessary rebuilds?

Comment (dwijnand) [reply src/rustup-dist/build.rs]: I was torn about whether to assume this was still necessary or whether it was legacy cruft.  Looks like I went with the latter assumption.  I'll recover it and we can dump it another day if need be.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**